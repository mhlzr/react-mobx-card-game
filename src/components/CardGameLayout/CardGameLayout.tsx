import React, { ReactElement, FunctionComponent, StatelessComponent } from 'react';
import styled from 'styled-components';
import { observer, inject } from 'mobx-react';

import { CardGameStore } from '../../stores/CardGameStore';

import { CardDetails } from '../CardDetails/CardDetails';
import { CardControls } from '../CardControls/CardControls';
import { CardOverview } from '../CardOverview/CardOverview';

const GRID_LAYOUT_WITH_SELECTED_PLAYER = `
    "details controls"
    "overview ."
`;

const GRID_LAYOUT_WITHOUT_SELECTED_PLAYER = `
    "overview controls"
`;

const Container = styled.div`
    display: grid;
    grid-gap: 1em;
    grid-template-columns: 2fr 1fr;
    grid-template-areas: ${({ player }) => player ? GRID_LAYOUT_WITH_SELECTED_PLAYER : GRID_LAYOUT_WITHOUT_SELECTED_PLAYER};
    max-width: 100em;
    padding: 1em;
    width: 100vw;
`;

interface CardGameLayoutProps {
    store: CardGameStore
}

export const CardGameLayout: FunctionComponent<CardGameLayoutProps> = inject('store')(observer(({ store }: CardGameLayoutProps): ReactElement => {
    return (
        <Container player={store.player}>
            <CardDetails />
            <CardControls />
            <CardOverview />
        </Container>)
}));
