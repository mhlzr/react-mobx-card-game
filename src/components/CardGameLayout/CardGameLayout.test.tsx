import React from 'react';
import { shallow } from 'enzyme'

import { CardGameLayoutComponent } from './CardGameLayout';

describe('CardGameLayout', () => {
    it('has a snapshot', () => {
        const wrapper = shallow(<CardGameLayoutComponent />);
        expect(wrapper).toMatchSnapshot();
    });
})
