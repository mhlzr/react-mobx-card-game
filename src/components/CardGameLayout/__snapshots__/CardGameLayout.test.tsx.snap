// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardGameLayout has a snapshot 1`] = `
<styled.div
  player={null}
>
  <inject-_class-with-store />
  <inject-_class-with-store />
  <inject-_class-with-store />
</styled.div>
`;
